main.c

```diff
@@ -134,7 +138,8 @@ main(int argc, char *argv[])
 
 	/* Prompt for a password. */
 	if (readpass(&passwd, "Please enter passphrase",
-	    dec ? NULL : "Please confirm passphrase", 1))
+	    dec ? NULL : !devtty ? NULL :
```

Comment:
Ok, this makes sense.  Sorry about the delay -- I wanted to check how I was using `readpass` in other code in case I wanted to change the non-tty behaviour, but I think it's best to leave that as reading twice even if we *think* we're not reading from a tty in case there's weirdness going on and we don't detect it correctly.

One small request, can you spell `dec ? NULL : !devtty ? NULL : ...` as `(dec || !devtty) ? NULL : ...` instead?  Seems both simpler and easier to understand.